Source/WebKit/UIProcess/WebPageProxy.h

```diff
@@ -1764,7 +1764,8 @@ class WebPageProxy final : public API::ObjectImpl<API::Object::Type::Page>, publ
     void requestFocusedElementInformation(CompletionHandler<void(const std::optional<FocusedElementInformation>&)>&&);
 
 #if PLATFORM(COCOA) || PLATFORM(GTK)
-    RefPtr<ViewSnapshot> takeViewSnapshot(std::optional<WebCore::IntRect>&&);
+    enum class ShouldAlwaysUseNonHardwareCapturePath : bool { No, Yes };
```

Comment:
Naming: I think 'ForceSoftwareRendering' or similar is better than 'ShouldAlwaysUseNonHardwareCapturePath'.